src/components/Home.jsx

```diff
@@ -1,19 +1,97 @@
-import React from 'react';
+import React, { useState } from 'react';
 import getToken from '../lib/tokens';
 import { useHistory } from 'react-router-dom';
+import db from '../lib/firebase';
+import firebase from 'firebase/app';
+
+import './Home.css';
 
 const Home = () => {
   let history = useHistory();
+  const [tokenName, setTokenName] = useState('');
+  const [notification, setNotification] = useState(null);
 
   const generateToken = () => {
     const token = getToken();
     localStorage.setItem('token', token);
+
+    db.collection('tokens')
+      .doc('tokenDocument')
+      .update({
+        tokenArray: firebase.firestore.FieldValue.arrayUnion(token),
+      })
+      .catch((error) => {
+        db.collection('tokens')
+          .doc('tokenDocument')
+          .set({
+            tokenArray: [token],
+          });
+      });
+
     history.push('/list-view');
   };
 
+  const resetError = () => {
+    setTimeout(() => {
+      setNotification(null);
+    }, 10000);
+  };
+
+  const compareToken = (e) => {
+    e.preventDefault();
+
+    db.collection('tokens')
+      .where('tokenArray', 'array-contains', tokenName)
+      .get()
+      .then((querySnapshot) => {
+        let tokenInDB;
+        querySnapshot.forEach((doc) => {
+          tokenInDB = doc.exists;
+        });
+        if (tokenInDB) {
+          localStorage.setItem('token', tokenName);
+          history.push('/list-view');
+        } else {
+          setNotification(
+            'Token not found - please check spelling and try again or create new list',
```

Comment:
I love love love these notification messages. Amazing work team 🥇 

If you want to take it further, you can add another notification message for an empty input field as well. 
For example, if a user does not enter any letters in the `share token` input field, you can display a message saying: "Please enter a token". This is just an idea 🙂

Reply:
Hi @Lukabaramishvili, thank you for the suggestions! 😊
Me and Claire decided to create separate array as a way of immediately storing the tokens to the DB on list creation, so that an empty list can be shared if the user wants to. I couldn't think of a way to do this with the items collection alone 🤔
Do you think it would be possible?

Reply:
It makes sense now 🙂. I was curious to know why you took this route.

Following the AC: "Entering the token and hitting submit saves the token to localStorage, effectively giving them joint control of **the list**"
I think the list is considered a list only if it has some items in it. Given this consideration, users will be able to share lists that have at least one item inside.

This is my personal observation, therefore please please feel free to take whichever route makes better sense to you 🙂

Please let me know if you have any clarifying questions 🙌

Reply:
It makes sense to me that an empty list should be able to exist regardless of how many items are in it, including zero. I know this isn't a traditional todo list, but consider what would happen if we completed all the items in a list (or removed them) which effectively removes any reference to the list in the database. No items would be able to be created. Storing a list of active lists enables this flexibility.

Reply:
That's a great observation and makes total sense. It's totally up to us to decide which route is the best 👍 
And we can always restructure the DB to make it better 🙌